src/inmanta/ast/statements/generator.py

```diff
@@ -143,7 +143,7 @@ def execute(self, requires: dict[object, object], instance: Instance, queue: Que
         try:
             inmanta_type.Bool().validate(condition)
         except RuntimeException as e:
-            e.set_statement(self.implements)
+            e.set_statement(self)
```

Comment:
Why is this?

Reply:
This was because of this test: https://github.com/inmanta/inmanta-core/blob/1be072d3347839804f8c598520089966c5cb73b5/tests/compiler/test_conditional.py#L483

```diff
+ ('Invalid value `some_string`: the condition for a conditional implementation '
+ 'can only be a boolean expression (reported in <inmanta.ast.entity.Implement '
+ 'object at 0x7f8be8da0a90> (/tmp/tmpq26mn7x6/main.cf:10:11))')
- ('Invalid value `some_string`: the condition for a conditional implementation '
- 'can only be a boolean expression (reported in implement __config__::A using '
- 'i when str (/tmp/tmpq26mn7x6/main.cf:10:11))')
```

Doing it this way might have some implications that I'm missing, the big picture is still not super clear to me.

Reply:
Fixed in a different way: don't set the statement in here, and instead set it one layer above through https://github.com/inmanta/inmanta-core/blob/b830762ba7e217e149bcac261765d45c4ee7787b/src/inmanta/execute/runtime.py#L1018
wdyt ?